### PR TITLE
Fix for 54448

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/54448
---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by spacedmonkey** [comment]
CC @dlh01

File: src/wp-includes/rest-api/endpoints/class-wp-rest-edit-site-export-controller.php [hunk 1]
```diff
@@ -54,7 +54,7 @@ public function register_routes() {
 	 */
 	public function permissions_check() {
 		if ( ! current_user_can( 'edit_theme_options' ) ) {
-			new WP_Error(
+			return new WP_Error(
 				'rest_cannot_view_url_details',
```
Comment (dlh01) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-edit-site-export-controller.php]:
```suggestion
				'rest_cannot_export_templates',
```

**Comment by spacedmonkey** [comment]
CC @Mamaduka

**Comment by Mamaduka** [comment]
Thanks for the follow-up, @spacedmonkey. The code looks good to me.

**Comment by TimothyBJacobs** [comment]
Great catch @dlh01, thanks for that!

This looks fine to me @spacedmonkey. If you wanted, we could consider instead making an `OPTIONS` request and checking the `Allow` header. That way, if the test fails, it doesn't kill the test suite execution because of the `die()` call. But I think this is fine as is.

File: tests/phpunit/tests/rest-api/wpRestEditSiteExportController.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	public function test_context_param() {}
+	public function test_get_item() {}
+	public function test_get_items() {}
+	public function test_create_item() {}
+	public function test_update_item() {}
+	public function test_delete_item() {}
+	public function test_prepare_item() {}
+	public function test_get_item_schema() {}
```
Comment (hellofromtonya) [inline tests/phpunit/tests/rest-api/wpRestEditSiteExportController.php]: These tests should marked as skip else they will be flagged as risky tests.
```suggestion
	public function test_context_param() {
		$this->markTestSkipped( 'Controller does not implement context_param().' );
	}

	public function test_get_item() {
		$this->markTestSkipped( 'Controller does not implement get_item().' );
	}

	public function test_get_items() {
		$this->markTestSkipped( 'Controller does not implement get_items().' );
	}

	public function test_create_item() {
		$this->markTestSkipped( 'Controller does not implement create_item().' );
	}

	public function test_update_item() {
		$this->markTestSkipped( 'Controller does not implement update_item().' );
	}

	public function test_delete_item() {
		$this->markTestSkipped( 'Controller does not implement delete_item().' );
	}

	public function test_prepare_item() {
		$this->markTestSkipped( 'Controller does not implement prepare_item().' );	
	}

	public function test_get_item_schema() {
		$this->markTestSkipped( 'Controller does not implement get_item_schema().' );
	}
```

The above suggestion assumes those methods are not implemented in the Controller. If they implemented, then change the `markTestSkipped()` to `markTestIncomplete()`.

**Pull request closed without merging**